### DISCUSSION TITLE
Benchmarks result?

### DISCUSSION THREAD

**Discussion by frederikhors** [cornucopia-rs/cornucopia#217] [Q&A] [answered]
Am I wrong in saying that I see the code for the sqlx/diesel benchmarks but I don't see the results?

Are they posted anywhere?

**Comment by LouisGariepy** [comment]
I don't think our benchmarks include SQLx (PR welcome!) although they do include Diesel and postgres/tokio-postgres.

The results are not published online since that would require periodically running benchmarks on a specialized, dedicated CI machine. Regular CI runners are unsuitable for benchmarking due to massive fluctuations in performance and general unreliability.

That's something we could implement in the future, but for the time being you can run the benchmarks on your local machine. It should be easy to run.

**Comment by LouisGariepy** [comment, marked as answer]
Generally speaking though, Cornucopia queries perform very close to `postgres` (for sync codegen) or `tokio-postgres` (for async codegen). Knowing this, you can compare using Diesel's benchmarking suite: https://github.com/diesel-rs/metrics
